app/javascript/shared/mixins/inboxMixin.js

```diff
@@ -36,12 +36,25 @@ export default {
       return this.channelType === INBOX_TYPES.EMAIL;
     },
     isATwilioSMSChannel() {
-      const { phone_number: phoneNumber = '' } = this.inbox;
-      return this.isATwilioChannel && !phoneNumber.startsWith('whatsapp');
+      const { medium: medium = '' } = this.inbox;
+      return this.isATwilioChannel && medium === 'sms';
     },
     isATwilioWhatsappChannel() {
-      const { phone_number: phoneNumber = '' } = this.inbox;
-      return this.isATwilioChannel && phoneNumber.startsWith('whatsapp');
+      const { medium: medium = '' } = this.inbox;
+      return this.isATwilioChannel && medium === 'whatsapp';
+    },
+    findInboxBadgeType() {
```

Comment:
Function `findInboxBadgeType` has a Cognitive Complexity of 6 (exceeds 5 allowed). Consider refactoring.